src/components/ResourceIcon.tsx

```diff
@@ -0,0 +1,47 @@
+import { Icon } from "@canonical/react-components";
+import { FC } from "react";
+
+export type ResourceIconType =
+  | "container"
+  | "virtual-machine"
+  | "snapshot"
+  | "profile"
+  | "project"
+  | "cluster-member"
+  | "network"
+  | "pool"
+  | "volume"
+  | "iso-volume"
+  | "image"
+  | "oidc-identity"
+  | "certificate"
+  | "auth-group"
+  | "device";
+
+const resourceIcons: Record<ResourceIconType, string> = {
+  container: "profiles",
+  "virtual-machine": "pods",
+  snapshot: "snapshot",
+  profile: "repository",
+  project: "folder",
+  "cluster-member": "single-host",
+  network: "exposed",
+  pool: "status-queued-small",
+  volume: "status-queued-small",
+  "iso-volume": "iso",
+  image: "image",
+  "oidc-identity": "user",
+  certificate: "certificate",
+  "auth-group": "user-group",
+  device: "units",
+};
+
+interface Props {
+  type: ResourceIconType;
+}
+
+const ResoruceIcon: FC<Props> = ({ type }) => {
```

Comment:
```suggestion
const ResourceIcon: FC<Props> = ({ type }) => {
```
still a typo :)

Reply:
The only one I didn't change is the original one you mentioned... :sweat_smile: